flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/sink/KafkaSink.java

```diff
@@ -132,4 +142,30 @@ public SimpleVersionedSerializer<KafkaWriterState> getWriterStateSerializer() {
     protected Properties getKafkaProducerConfig() {
         return kafkaProducerConfig;
     }
+
+    @Override
+    public LineageVertex getLineageVertex() {
+        // enrich dataset facet with properties
+        if (recordSerializer instanceof KafkaDatasetFacetProvider) {
+            Optional<KafkaDatasetFacet> kafkaDatasetFacet =
+                    ((KafkaDatasetFacetProvider) recordSerializer).getKafkaDatasetFacet();
+
+            if (!kafkaDatasetFacet.isPresent()) {
+                LOG.warn("Provided did not return kafka dataset facet");
+                return null;
+            }
+
+            kafkaDatasetFacet.get().addProperties(this.kafkaProducerConfig);
```

Comment:
Do we ever actually get the properties from the recordSerializer? So are we actually just setting here?

Reply:
ok, we can convert it to `setter`.